File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int size = heights.size();
        int left = 0, right = size - 1, max = 0;
        int tmp;
        while (left < right)
        {
            tmp =  (heights[left] < heights[right] ? heights[left] : heights[right]) * (right - left);
            max = tmp > max ? tmp : max;
            if (heights[left] < heights[right])
                left++;
            else
                right--;
        }
        return max;
    }
};

File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-0.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        stack = []
        heights.append(0)
        max_area = 0
        for i, h in enumerate(heights):
            while stack and heights[stack[-1]] > h:
                hight = heights[stack.pop()]
                width = i if not stack else i - stack[-1] - 1
                max_area = max(max_area, hight * width)
            stack.append(i)
        return max_area
